src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5048,34 +5049,141 @@ namespace winrt::TerminalApp::implementation
 
             button.Label(label);
             button.Click(makeCallback(action));
-            menu.SecondaryCommands().Append(button);
+            targetMenu.SecondaryCommands().Append(button);
         };
 
+        auto makeMenuItem = [](const winrt::hstring& label,
+                               const winrt::hstring& icon,
+                               const auto& subMenu,
+                               auto& targetMenu) {
+            AppBarButton button{};
+
+            if (!icon.empty())
+            {
+                auto iconElement = UI::IconPathConverter::IconWUX(icon);
+                Automation::AutomationProperties::SetAccessibilityView(iconElement, Automation::Peers::AccessibilityView::Raw);
+                button.Icon(iconElement);
+            }
+
+            button.Label(label);
+            button.Flyout(subMenu);
+            targetMenu.SecondaryCommands().Append(button);
+        };
+
+        const auto focusedProfile = _GetFocusedTabImpl()->GetFocusedProfile();
+        auto separatorItem = AppBarSeparator{};
+        auto activeProfiles = _settings.ActiveProfiles();
+        auto activeProfileCount = gsl::narrow_cast<int>(activeProfiles.Size());
+        MUX::Controls::CommandBarFlyout splitPaneDownMenu{};
+        MUX::Controls::CommandBarFlyout splitPaneUpMenu{};
+        MUX::Controls::CommandBarFlyout splitPaneRightMenu{};
+        MUX::Controls::CommandBarFlyout splitPaneLeftMenu{};
+
         // Wire up each item to the action that should be performed. By actually
         // connecting these to actions, we ensure the implementation is
         // consistent. This also leaves room for customizing this menu with
         // actions in the future.
 
-        makeItem(RS_(L"SplitPaneText"), L"\xF246", ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Duplicate } });
-        makeItem(RS_(L"DuplicateTabText"), L"\xF5ED", ActionAndArgs{ ShortcutAction::DuplicateTab, nullptr });
+        if (!withSelection)
+        {
+            makeItem(RS_(L"FindText"), L"\xF78B", ActionAndArgs{ ShortcutAction::Find, nullptr }, menu);
```

Comment:
Interesting - this would _always_ add the "Find..." item, because the control only requests it when there _is_ a selection. 

Maybe we should just have the control always do this itself?